sqlite/sqlite.go

```diff
@@ -230,14 +231,38 @@ func (s *SqlStore) RestoreSqlStore(ctx context.Context, r io.Reader) error {
 		return err
 	}
 
-	// Close the current DB.
+	// Close the current DB and run the restore while under lock.
+	s.Mu.Lock()
```

Comment:
Didn't have locks here before but it seems like we really should be getting a lock prior to doing the sqlite database restoration as well so I adapted similar logic as in the bolt kv restore code for locking and unlocking.